// clarity-angular/popover/signpost/signpost.ngfactory.js
/**
 * @fileoverview This file is generated by the Angular template compiler.
 * Do not edit.
 * @suppress {suspiciousCode,uselessCode,missingProperties,missingOverride}
 */
/* tslint:disable */
import * as i0 from '@angular/core';
import * as i1 from 'clarity-angular/popover/signpost/signpost-trigger.directive';
import * as i2 from 'clarity-angular/utils/conditional/if-open.service';
import * as i3 from 'clarity-angular/icon/icon';
import * as i4 from '@angular/common';
import * as i5 from 'clarity-angular/popover/signpost/signpost';
var styles_Signpost = [];
export var RenderType_Signpost = i0.ɵcrt({ encapsulation: 2, styles: styles_Signpost,
    data: {} });
function View_Signpost_1(_l) {
    return i0.ɵvid(0, [(_l()(), i0.ɵeld(0, null, null, 8, null, null, null, null, null, null, null)),
        (_l()(), i0.ɵted(null, ['\n            '])), (_l()(), i0.ɵeld(0, null, null, 5, 'button', [['class', 'signpost-action btn btn-small btn-link'],
            ['clrSignpostTrigger', ''], ['type', 'button']], null, [[null,
                'click']], function (_v, en, $event) {
            var ad = true;
            if (('click' === en)) {
                var pd_0 = (i0.ɵnov(_v, 3).onSignpostTriggerClick($event) !== false);
                ad = (pd_0 && ad);
            }
            return ad;
        }, null, null)), i0.ɵdid(16384, null, 0, i1.SignpostTriggerDirective, [i2.IfOpenService], null, null), (_l()(), i0.ɵted(null, ['\n                '])), (_l()(), i0.ɵeld(0, null, null, 1, 'clr-icon', [['shape', 'info']], null, null, null, null, null)), i0.ɵdid(16384, null, 0, i3.IconCustomTag, [], null, null), (_l()(), i0.ɵted(null, ['\n            '])), (_l()(), i0.ɵted(null, ['\n        ']))], null, null);
}
export function View_Signpost_0(_l) {
    return i0.ɵvid(0, [(_l()(), i0.ɵted(null, ['\n        '])), (_l()(), i0.ɵand(16777216, null, null, 1, null, View_Signpost_1)), i0.ɵdid(16384, null, 0, i4.NgIf, [i0.ViewContainerRef, i0.TemplateRef], { ngIf: [0, 'ngIf'] }, null), (_l()(), i0.ɵted(null, ['\n        \n        '])), i0.ɵncd(null, 0), (_l()(), i0.ɵted(null, ['\n    ']))], function (_ck, _v) {
        var _co = _v.component;
        var currVal_0 = !_co.useCustomTrigger;
        _ck(_v, 2, 0, currVal_0);
    }, null);
}
export function View_Signpost_Host_0(_l) {
    return i0.ɵvid(0, [(_l()(), i0.ɵeld(0, null, null, 3, 'clr-signpost', [], [[2, 'signpost', null]], null, null, View_Signpost_0, RenderType_Signpost)), i0.ɵprd(4608, null, i2.IfOpenService, i2.IfOpenService, []), i0.ɵdid(49152, null, 1, i5.Signpost, [], null, null), i0.ɵqud(335544320, 1, { customTrigger: 0 })], null, function (_ck, _v) {
        var currVal_0 = true;
        _ck(_v, 0, 0, currVal_0);
    });
}
export var SignpostNgFactory = i0.ɵccf('clr-signpost', i5.Signpost, View_Signpost_Host_0, {}, {}, ['*']);
