// clarity-angular/utils/template-ref/template-ref-container.ngfactory.js
/**
 * @fileoverview This file is generated by the Angular template compiler.
 * Do not edit.
 * @suppress {suspiciousCode,uselessCode,missingProperties,missingOverride}
 */
/* tslint:disable */
import * as i0 from '@angular/core';
import * as i1 from 'clarity-angular/utils/template-ref/template-ref-container';
var styles_TemplateRefContainer = [];
export var RenderType_TemplateRefContainer = i0.ɵcrt({ encapsulation: 2,
    styles: styles_TemplateRefContainer, data: {} });
function View_TemplateRefContainer_1(_l) {
    return i0.ɵvid(0, [(_l()(), i0.ɵted(null, ['\n        '])), i0.ɵncd(null, 0), (_l()(), i0.ɵted(null, ['\n      ']))], null, null);
}
export function View_TemplateRefContainer_0(_l) {
    return i0.ɵvid(0, [i0.ɵqud(402653184, 1, { template: 0 }), (_l()(), i0.ɵted(null, ['\n      '])), (_l()(), i0.ɵand(0, [[1, 2]], null, 0, null, View_TemplateRefContainer_1)),
        (_l()(), i0.ɵted(null, ['\n    ']))], null, null);
}
export function View_TemplateRefContainer_Host_0(_l) {
    return i0.ɵvid(0, [(_l()(), i0.ɵeld(0, null, null, 1, 'ng-component', [], null, null, null, View_TemplateRefContainer_0, RenderType_TemplateRefContainer)), i0.ɵdid(49152, null, 0, i1.TemplateRefContainer, [], null, null)], null, null);
}
export var TemplateRefContainerNgFactory = i0.ɵccf('ng-component', i1.TemplateRefContainer, View_TemplateRefContainer_Host_0, {}, {}, ['*']);
