// clarity-angular/layout/vertical-nav/vertical-nav-group.ngfactory.js
/**
 * @fileoverview This file is generated by the Angular template compiler.
 * Do not edit.
 * @suppress {suspiciousCode,uselessCode,missingProperties,missingOverride}
 */
/* tslint:disable */
import * as i0 from '@angular/core';
import * as i1 from 'clarity-angular/layout/vertical-nav/vertical-nav-group';
import * as i2 from 'clarity-angular/icon/icon';
import * as i3 from 'clarity-angular/utils/expand/providers/expand';
import * as i4 from 'clarity-angular/layout/vertical-nav/providers/vertical-nav-group.service';
import * as i5 from 'clarity-angular/layout/vertical-nav/providers/vertical-nav.service';
var styles_VerticalNavGroup = [];
export var RenderType_VerticalNavGroup = i0.ɵcrt({ encapsulation: 2,
    styles: styles_VerticalNavGroup, data: { 'animation': [{ type: 7, name: 'clrExpand', definitions: [{ type: 0,
                        name: 'expanded', styles: { type: 6, styles: { 'height': '*' }, offset: null },
                        options: undefined }, { type: 0, name: 'collapsed', styles: { type: 6, styles: { 'height': 0,
                                'overflow-y': 'hidden' }, offset: null }, options: undefined }, { type: 1,
                        expr: 'expanded <=> collapsed', animation: { type: 4, styles: null, timings: '0.2s ease-in-out' },
                        options: null }], options: {} }] } });
export function View_VerticalNavGroup_0(_l) {
    return i0.ɵvid(0, [(_l()(), i0.ɵted(null, ['\n      '])), (_l()(), i0.ɵeld(0, null, null, 16, 'div', [['class', 'nav-group-content']], null, [[null, 'click']], function (_v, en, $event) {
            var ad = true;
            var _co = _v.component;
            if (('click' === en)) {
                var pd_0 = (_co.onNavContentClick() !== false);
                ad = (pd_0 && ad);
            }
            return ad;
        }, null, null)), (_l()(), i0.ɵted(null, ['\n          '])),
        i0.ɵncd(null, 0), (_l()(), i0.ɵted(null, ['\n          '])), i0.ɵncd(null, 1), (_l()(), i0.ɵted(null, ['\n          '])), (_l()(), i0.ɵeld(0, null, null, 3, 'div', [['class', 'nav-group-text']], null, null, null, null, null)), (_l()(), i0.ɵted(null, ['\n              '])), i0.ɵncd(null, 2), (_l()(), i0.ɵted(null, ['\n          '])), (_l()(), i0.ɵted(null, ['\n          '])), (_l()(),
            i0.ɵeld(0, null, null, 4, 'button', [['class', 'nav-group-trigger'],
                ['type', 'button']], null, [[null, 'click']], function (_v, en, $event) {
                var ad = true;
                var _co = _v.component;
                if (('click' === en)) {
                    var pd_0 = (_co.toggleExpand() !== false);
                    ad = (pd_0 && ad);
                }
                return ad;
            }, null, null)), (_l()(), i0.ɵted(null, ['\n              '])),
        (_l()(), i0.ɵeld(0, null, null, 1, 'clr-icon', [['shape', 'caret']], [[1, 'dir', 0]], null, null, null, null)),
        i0.ɵdid(16384, null, 0, i2.IconCustomTag, [], null, null),
        (_l()(), i0.ɵted(null, ['\n          '])), (_l()(), i0.ɵted(null, ['\n      '])), (_l()(), i0.ɵted(null, ['\n      '])), (_l()(), i0.ɵted(null, ['\n      '])), (_l()(), i0.ɵeld(0, null, null, 3, 'div', [['class',
                'nav-group-children']], [[24, '@clrExpand', 0]], [[null, '@clrExpand.done']], function (_v, en, $event) {
            var ad = true;
            var _co = _v.component;
            if (('@clrExpand.done' === en)) {
                var pd_0 = (_co.expandAnimationDone($event) !== false);
                ad = (pd_0 && ad);
            }
            return ad;
        }, null, null)), (_l()(), i0.ɵted(null, ['\n          '])),
        i0.ɵncd(null, 3), (_l()(), i0.ɵted(null, ['\n      '])), (_l()(),
            i0.ɵted(null, ['\n    ']))], null, function (_ck, _v) {
        var _co = _v.component;
        var currVal_0 = (_co.expanded ? 'down' : 'right');
        _ck(_v, 14, 0, currVal_0);
        var currVal_1 = _co.expandAnimationState;
        _ck(_v, 20, 0, currVal_1);
    });
}
export function View_VerticalNavGroup_Host_0(_l) {
    return i0.ɵvid(0, [(_l()(), i0.ɵeld(0, null, null, 2, 'clr-vertical-nav-group', [['class', 'nav-group']], [[2, 'is-expanded', null]], null, null, View_VerticalNavGroup_0, RenderType_VerticalNavGroup)), i0.ɵprd(512, null, i3.Expand, i3.Expand, []), i0.ɵdid(1228800, null, 0, i1.VerticalNavGroup, [i3.Expand, i4.VerticalNavGroupService, i5.VerticalNavService], null, null)], null, function (_ck, _v) {
        var currVal_0 = i0.ɵnov(_v, 2).expanded;
        _ck(_v, 0, 0, currVal_0);
    });
}
export var VerticalNavGroupNgFactory = i0.ɵccf('clr-vertical-nav-group', i1.VerticalNavGroup, View_VerticalNavGroup_Host_0, { userExpandedInput: 'clrVerticalNavGroupExpanded' }, { expandedChange: 'clrVerticalNavGroupExpandedChange' }, ['[clrVerticalNavIcon]', '[clrVerticalNavLink]',
    '*', '[clrIfExpanded], clr-vertical-nav-group-children']);
