// clarity-angular/layout/vertical-nav/vertical-nav.ngfactory.js
/**
 * @fileoverview This file is generated by the Angular template compiler.
 * Do not edit.
 * @suppress {suspiciousCode,uselessCode,missingProperties,missingOverride}
 */
/* tslint:disable */
import * as i0 from '@angular/core';
import * as i1 from 'clarity-angular/icon/icon';
import * as i2 from '@angular/common';
import * as i3 from 'clarity-angular/layout/vertical-nav/vertical-nav';
import * as i4 from 'clarity-angular/layout/vertical-nav/providers/vertical-nav.service';
import * as i5 from 'clarity-angular/layout/vertical-nav/providers/vertical-nav-icon.service';
import * as i6 from 'clarity-angular/layout/vertical-nav/providers/vertical-nav-group.service';
var styles_VerticalNav = [];
export var RenderType_VerticalNav = i0.ɵcrt({ encapsulation: 2, styles: styles_VerticalNav,
    data: {} });
function View_VerticalNav_1(_l) {
    return i0.ɵvid(0, [(_l()(), i0.ɵeld(0, null, null, 4, 'button', [['class',
                'nav-trigger'], ['type', 'button']], [[2, 'on-collapse', null]], [[null,
                'click']], function (_v, en, $event) {
            var ad = true;
            var _co = _v.component;
            if (('click' === en)) {
                var pd_0 = (_co.toggleByButton() !== false);
                ad = (pd_0 && ad);
            }
            return ad;
        }, null, null)), (_l()(), i0.ɵted(null, ['\n          '])),
        (_l()(), i0.ɵeld(0, null, null, 1, 'clr-icon', [['class', 'nav-trigger-icon'],
            ['shape', 'angle-double']], [[1, 'dir', 0]], null, null, null, null)), i0.ɵdid(16384, null, 0, i1.IconCustomTag, [], null, null), (_l()(), i0.ɵted(null, ['\n      ']))], null, function (_ck, _v) {
        var _co = _v.component;
        var currVal_0 = _co.collapsed;
        _ck(_v, 0, 0, currVal_0);
        var currVal_1 = (_co.collapsed ? 'right' : 'left');
        _ck(_v, 2, 0, currVal_1);
    });
}
function View_VerticalNav_2(_l) {
    return i0.ɵvid(0, [(_l()(), i0.ɵeld(0, null, null, 0, 'button', [['class',
                'nav-btn']], null, [[null, 'click']], function (_v, en, $event) {
            var ad = true;
            var _co = _v.component;
            if (('click' === en)) {
                var pd_0 = ((_co.collapsed = false) !== false);
                ad = (pd_0 && ad);
            }
            return ad;
        }, null, null))], null, null);
}
export function View_VerticalNav_0(_l) {
    return i0.ɵvid(0, [(_l()(), i0.ɵted(null, ['\n      '])), (_l()(), i0.ɵand(16777216, null, null, 1, null, View_VerticalNav_1)), i0.ɵdid(16384, null, 0, i2.NgIf, [i0.ViewContainerRef, i0.TemplateRef], { ngIf: [0, 'ngIf'] }, null), (_l()(), i0.ɵted(null, ['\n      '])), (_l()(), i0.ɵted(null, ['\n      '])), (_l()(), i0.ɵeld(0, null, null, 6, 'div', [['class',
                'nav-content']], null, null, null, null, null)),
        (_l()(), i0.ɵted(null, ['\n          '])), i0.ɵncd(null, 0), (_l()(),
            i0.ɵted(null, ['\n          '])), (_l()(), i0.ɵand(16777216, null, null, 1, null, View_VerticalNav_2)), i0.ɵdid(16384, null, 0, i2.NgIf, [i0.ViewContainerRef, i0.TemplateRef], { ngIf: [0, 'ngIf'] }, null),
        (_l()(), i0.ɵted(null, ['\n      '])), (_l()(), i0.ɵted(null, ['\n    ']))], function (_ck, _v) {
        var _co = _v.component;
        var currVal_0 = _co.collapsible;
        _ck(_v, 2, 0, currVal_0);
        var currVal_1 = (_co.collapsible && _co.collapsed);
        _ck(_v, 10, 0, currVal_1);
    }, null);
}
export function View_VerticalNav_Host_0(_l) {
    return i0.ɵvid(0, [(_l()(), i0.ɵeld(0, null, null, 4, 'clr-vertical-nav', [['class', 'clr-vertical-nav']], [[2, 'is-collapsed', null], [2, 'has-nav-groups',
                null], [2, 'has-icons', null]], null, null, View_VerticalNav_0, RenderType_VerticalNav)), i0.ɵprd(512, null, i4.VerticalNavService, i4.VerticalNavService, []), i0.ɵprd(512, null, i5.VerticalNavIconService, i5.VerticalNavIconService, []), i0.ɵprd(512, null, i6.VerticalNavGroupService, i6.VerticalNavGroupService, []), i0.ɵdid(180224, null, 0, i3.VerticalNav, [i4.VerticalNavService, i5.VerticalNavIconService, i6.VerticalNavGroupService], null, null)], null, function (_ck, _v) {
        var currVal_0 = i0.ɵnov(_v, 4).collapsed;
        var currVal_1 = i0.ɵnov(_v, 4).hasNavGroups;
        var currVal_2 = i0.ɵnov(_v, 4).hasIcons;
        _ck(_v, 0, 0, currVal_0, currVal_1, currVal_2);
    });
}
export var VerticalNavNgFactory = i0.ɵccf('clr-vertical-nav', i3.VerticalNav, View_VerticalNav_Host_0, { collapsible: 'clrVerticalNavCollapsible',
    collapsed: 'clrVerticalNavCollapsed' }, { _collapsedChanged: 'clrVerticalNavCollapsedChange' }, ['*']);
