// clarity-angular/wizard/wizard.ngfactory.js
/**
 * @fileoverview This file is generated by the Angular template compiler.
 * Do not edit.
 * @suppress {suspiciousCode,uselessCode,missingProperties,missingOverride}
 */
/* tslint:disable */
import * as i0 from '@angular/core';
import * as i1 from '@angular/common';
import * as i2 from 'clarity-angular/wizard/wizard';
import * as i3 from '../modal/modal.ngfactory';
import * as i4 from 'clarity-angular/utils/scrolling/scrolling-service';
import * as i5 from '@angular/platform-browser';
import * as i6 from 'clarity-angular/modal/modal';
import * as i7 from './wizard-stepnav.ngfactory';
import * as i8 from 'clarity-angular/wizard/wizard-stepnav';
import * as i9 from 'clarity-angular/wizard/providers/page-collection';
import * as i10 from 'clarity-angular/wizard/providers/button-hub';
import * as i11 from 'clarity-angular/wizard/providers/wizard-navigation';
import * as i12 from 'clarity-angular/wizard/providers/header-actions';
var styles_Wizard = [];
export var RenderType_Wizard = i0.ɵcrt({ encapsulation: 2, styles: styles_Wizard,
    data: {} });
function View_Wizard_1(_l) {
    return i0.ɵvid(0, [(_l()(), i0.ɵand(0, null, null, 0))], null, null);
}
function View_Wizard_3(_l) {
    return i0.ɵvid(0, [(_l()(), i0.ɵeld(0, null, null, 3, 'div', [], null, null, null, null, null)), (_l()(),
            i0.ɵted(null, ['\n                      '])), i0.ɵncd(null, 1),
        (_l()(), i0.ɵted(null, ['\n                  ']))], null, null);
}
function View_Wizard_5(_l) {
    return i0.ɵvid(0, [(_l()(), i0.ɵand(0, null, null, 0))], null, null);
}
function View_Wizard_4(_l) {
    return i0.ɵvid(0, [(_l()(), i0.ɵeld(0, null, null, 4, 'div', [], null, null, null, null, null)), (_l()(),
            i0.ɵted(null, ['\n                      '])), (_l()(), i0.ɵand(16777216, null, null, 1, null, View_Wizard_5)), i0.ɵdid(540672, null, 0, i1.NgTemplateOutlet, [i0.ViewContainerRef], { ngTemplateOutlet: [0, 'ngTemplateOutlet'] }, null), (_l()(), i0.ɵted(null, ['\n                  ']))], function (_ck, _v) {
        var _co = _v.component;
        var currVal_0 = _co.navService.currentPage.headerActions;
        _ck(_v, 3, 0, currVal_0);
    }, null);
}
function View_Wizard_2(_l) {
    return i0.ɵvid(0, [(_l()(), i0.ɵeld(0, null, null, 7, 'div', [['class',
                'modal-header-actions-wrapper']], null, null, null, null, null)), (_l()(), i0.ɵted(null, ['\n                  '])), (_l()(),
            i0.ɵand(16777216, null, null, 1, null, View_Wizard_3)),
        i0.ɵdid(16384, null, 0, i1.NgIf, [i0.ViewContainerRef, i0.TemplateRef], { ngIf: [0,
                'ngIf'] }, null), (_l()(), i0.ɵted(null, ['\n                  '])),
        (_l()(), i0.ɵand(16777216, null, null, 1, null, View_Wizard_4)),
        i0.ɵdid(16384, null, 0, i1.NgIf, [i0.ViewContainerRef, i0.TemplateRef], { ngIf: [0,
                'ngIf'] }, null), (_l()(), i0.ɵted(null, ['\n              ']))], function (_ck, _v) {
        var _co = _v.component;
        var currVal_0 = _co.headerActionService.showWizardHeaderActions;
        _ck(_v, 3, 0, currVal_0);
        var currVal_1 = _co.headerActionService.currentPageHasHeaderActions;
        _ck(_v, 6, 0, currVal_1);
    }, null);
}
function View_Wizard_6(_l) {
    return i0.ɵvid(0, [(_l()(), i0.ɵeld(0, null, null, 3, 'div', [['class',
                'clr-wizard-footer-buttons-wrapper']], null, null, null, null, null)), (_l()(), i0.ɵted(null, ['\n                      '])),
        i0.ɵncd(null, 3), (_l()(), i0.ɵted(null, ['\n                  ']))], null, null);
}
function View_Wizard_8(_l) {
    return i0.ɵvid(0, [(_l()(), i0.ɵand(0, null, null, 0))], null, null);
}
function View_Wizard_7(_l) {
    return i0.ɵvid(0, [(_l()(), i0.ɵeld(0, null, null, 4, 'div', [['class',
                'clr-wizard-footer-buttons-wrapper']], null, null, null, null, null)), (_l()(), i0.ɵted(null, ['\n                      '])),
        (_l()(), i0.ɵand(16777216, null, null, 1, null, View_Wizard_8)),
        i0.ɵdid(540672, null, 0, i1.NgTemplateOutlet, [i0.ViewContainerRef], { ngTemplateOutlet: [0,
                'ngTemplateOutlet'] }, null), (_l()(), i0.ɵted(null, ['\n                  ']))], function (_ck, _v) {
        var _co = _v.component;
        var currVal_0 = _co.navService.currentPage.buttons;
        _ck(_v, 3, 0, currVal_0);
    }, null);
}
export function View_Wizard_0(_l) {
    return i0.ɵvid(0, [(_l()(), i0.ɵted(null, ['\n      '])), (_l()(), i0.ɵted(null, ['\n\n      '])), (_l()(), i0.ɵeld(0, null, null, 44, 'clr-modal', [], null, [[null, 'clrModalAlternateClose'], ['body',
                'keyup.escape']], function (_v, en, $event) {
            var ad = true;
            var _co = _v.component;
            if (('body:keyup.escape' === en)) {
                var pd_0 = (i0.ɵnov(_v, 4).close() !== false);
                ad = (pd_0 && ad);
            }
            if (('clrModalAlternateClose' === en)) {
                var pd_1 = (_co.modalCancel() !== false);
                ad = (pd_1 && ad);
            }
            return ad;
        }, i3.View_Modal_0, i3.RenderType_Modal)), i0.ɵprd(8704, null, i4.ScrollingService, i4.ScrollingService, [i5.DOCUMENT]), i0.ɵdid(704512, null, 0, i6.Modal, [i4.ScrollingService], { _open: [0, '_open'], closable: [1, 'closable'], size: [2, 'size'], staticBackdrop: [3,
                'staticBackdrop'], skipAnimation: [4, 'skipAnimation'], ghostPageState: [5, 'ghostPageState'],
            bypassScrollService: [6, 'bypassScrollService'], stopClose: [7, 'stopClose'] }, { altClose: 'clrModalAlternateClose' }), (_l()(), i0.ɵted(null, ['\n\n          '])),
        (_l()(), i0.ɵeld(0, null, 0, 7, 'nav', [['class', 'modal-nav clr-wizard-stepnav-wrapper']], null, null, null, null, null)),
        (_l()(), i0.ɵted(null, ['\n              '])), (_l()(), i0.ɵeld(0, null, null, 1, 'h3', [['class', 'clr-wizard-title']], null, null, null, null, null)), i0.ɵncd(null, 0), (_l()(),
            i0.ɵted(null, ['\n              '])), (_l()(), i0.ɵeld(0, null, null, 1, 'clr-wizard-stepnav', [['class', 'clr-wizard-stepnav']], null, null, null, i7.View_WizardStepnav_0, i7.RenderType_WizardStepnav)),
        i0.ɵdid(49152, null, 0, i8.WizardStepnav, [i9.PageCollectionService], null, null), (_l()(), i0.ɵted(null, ['\n          '])), (_l()(),
            i0.ɵted(null, ['\n\n          '])), (_l()(), i0.ɵeld(0, null, 1, 10, 'h3', [['class', 'modal-title']], null, null, null, null, null)), (_l()(), i0.ɵted(null, ['\n              '])),
        (_l()(), i0.ɵeld(0, null, null, 4, 'span', [['class', 'modal-title-text']], null, null, null, null, null)),
        (_l()(), i0.ɵted(null, ['\n                  '])), (_l()(), i0.ɵand(16777216, null, null, 1, null, View_Wizard_1)), i0.ɵdid(540672, null, 0, i1.NgTemplateOutlet, [i0.ViewContainerRef], { ngTemplateOutlet: [0,
                'ngTemplateOutlet'] }, null), (_l()(), i0.ɵted(null, ['\n              '])),
        (_l()(), i0.ɵted(null, ['\n\n              '])), (_l()(), i0.ɵand(16777216, null, null, 1, null, View_Wizard_2)), i0.ɵdid(16384, null, 0, i1.NgIf, [i0.ViewContainerRef, i0.TemplateRef], { ngIf: [0, 'ngIf'] }, null), (_l()(), i0.ɵted(null, ['\n          '])), (_l()(),
            i0.ɵted(null, ['\n\n          '])), (_l()(), i0.ɵeld(0, null, 2, 6, 'div', [['class', 'modal-body']], null, null, null, null, null)), (_l()(), i0.ɵted(null, ['\n              '])),
        (_l()(), i0.ɵeld(0, null, null, 3, 'main', [['class', 'clr-wizard-content'],
            ['clr-wizard-pages-wrapper', '']], null, null, null, null, null)), (_l()(), i0.ɵted(null, ['\n                  '])),
        i0.ɵncd(null, 2), (_l()(), i0.ɵted(null, ['\n              '])),
        (_l()(), i0.ɵted(null, ['\n          '])), (_l()(), i0.ɵted(null, ['\n          '])), (_l()(), i0.ɵeld(0, null, 3, 10, 'div', [['class',
                'modal-footer clr-wizard-footer']], null, null, null, null, null)), (_l()(), i0.ɵted(null, ['\n              '])),
        (_l()(), i0.ɵeld(0, null, null, 7, 'div', [['class', 'clr-wizard-footer-buttons']], null, null, null, null, null)),
        (_l()(), i0.ɵted(null, ['\n                  '])), (_l()(), i0.ɵand(16777216, null, null, 1, null, View_Wizard_6)), i0.ɵdid(16384, null, 0, i1.NgIf, [i0.ViewContainerRef, i0.TemplateRef], { ngIf: [0, 'ngIf'] }, null), (_l()(), i0.ɵted(null, ['\n                  '])),
        (_l()(), i0.ɵand(16777216, null, null, 1, null, View_Wizard_7)),
        i0.ɵdid(16384, null, 0, i1.NgIf, [i0.ViewContainerRef, i0.TemplateRef], { ngIf: [0,
                'ngIf'] }, null), (_l()(), i0.ɵted(null, ['\n              '])),
        (_l()(), i0.ɵted(null, ['\n          '])), (_l()(), i0.ɵted(null, ['\n      '])), (_l()(), i0.ɵted(null, ['\n    ']))], function (_ck, _v) {
        var _co = _v.component;
        var currVal_0 = _co._open;
        var currVal_1 = _co.closable;
        var currVal_2 = _co.size;
        var currVal_3 = true;
        var currVal_4 = _co.stopModalAnimations;
        var currVal_5 = _co.ghostPageState;
        var currVal_6 = _co.isStatic;
        var currVal_7 = true;
        _ck(_v, 4, 0, currVal_0, currVal_1, currVal_2, currVal_3, currVal_4, currVal_5, currVal_6, currVal_7);
        var currVal_8 = _co.navService.currentPageTitle;
        _ck(_v, 20, 0, currVal_8);
        var currVal_9 = _co.headerActionService.displayHeaderActionsWrapper;
        _ck(_v, 24, 0, currVal_9);
        var currVal_10 = (_co.navService.currentPage && !_co.navService.currentPage.hasButtons);
        _ck(_v, 40, 0, currVal_10);
        var currVal_11 = (_co.navService.currentPage && _co.navService.currentPage.hasButtons);
        _ck(_v, 43, 0, currVal_11);
    }, null);
}
export function View_Wizard_Host_0(_l) {
    return i0.ɵvid(0, [(_l()(), i0.ɵeld(0, null, null, 7, 'clr-wizard', [], [[2, 'clr-wizard', null], [2, 'wizard-md', null], [2, 'wizard-lg',
                null], [2, 'wizard-xl', null], [2, 'lastPage', null],
            [2, 'clr-wizard--ghosted', null]], null, null, View_Wizard_0, RenderType_Wizard)), i0.ɵprd(512, null, i9.PageCollectionService, i9.PageCollectionService, []), i0.ɵprd(512, null, i10.ButtonHubService, i10.ButtonHubService, []), i0.ɵprd(131584, null, i11.WizardNavigationService, i11.WizardNavigationService, [i9.PageCollectionService, i10.ButtonHubService]), i0.ɵprd(512, null, i12.HeaderActionService, i12.HeaderActionService, [i11.WizardNavigationService]), i0.ɵdid(1556480, null, 2, i2.Wizard, [i11.WizardNavigationService, i9.PageCollectionService, i10.ButtonHubService,
            i12.HeaderActionService, i0.ElementRef, i0.IterableDiffers], null, null), i0.ɵqud(603979776, 1, { pages: 1 }), i0.ɵqud(603979776, 2, { headerActions: 1 })], function (_ck, _v) {
        _ck(_v, 5, 0);
    }, function (_ck, _v) {
        var currVal_0 = true;
        var currVal_1 = (i0.ɵnov(_v, 5).size == 'md');
        var currVal_2 = (i0.ɵnov(_v, 5).size == 'lg');
        var currVal_3 = (i0.ɵnov(_v, 5).size == 'xl');
        var currVal_4 = i0.ɵnov(_v, 5).navService.currentPageIsLast;
        var currVal_5 = i0.ɵnov(_v, 5).showGhostPages;
        _ck(_v, 0, 0, currVal_0, currVal_1, currVal_2, currVal_3, currVal_4, currVal_5);
    });
}
export var WizardNgFactory = i0.ɵccf('clr-wizard', i2.Wizard, View_Wizard_Host_0, { size: 'clrWizardSize', showGhostPages: 'clrWizardShowGhostPages',
    forceForward: 'clrWizardForceForwardNavigation', closable: 'clrWizardClosable',
    _open: 'clrWizardOpen', stopNext: 'clrWizardPreventDefaultNext', stopCancel: 'clrWizardPreventDefaultCancel',
    stopNavigation: 'clrWizardPreventNavigation', disableStepnav: 'clrWizardDisableStepnav',
    _stopModalAnimations: 'clrWizardPreventModalAnimation' }, { _openChanged: 'clrWizardOpenChange',
    onCancel: 'clrWizardOnCancel', wizardFinished: 'clrWizardOnFinish', onReset: 'clrWizardOnReset',
    currentPageChanged: 'clrWizardCurrentPageChanged', onMoveNext: 'clrWizardOnNext',
    onMovePrevious: 'clrWizardOnPrevious' }, ['clr-wizard-title', 'clr-wizard-header-action',
    '*', 'clr-wizard-button']);
