// clarity-angular/wizard/wizard-stepnav-item.ngfactory.js
/**
 * @fileoverview This file is generated by the Angular template compiler.
 * Do not edit.
 * @suppress {suspiciousCode,uselessCode,missingProperties,missingOverride}
 */
/* tslint:disable */
import * as i0 from '@angular/core';
import * as i1 from 'clarity-angular/wizard/wizard-stepnav-item';
import * as i2 from '@angular/common';
import * as i3 from 'clarity-angular/wizard/providers/wizard-navigation';
import * as i4 from 'clarity-angular/wizard/providers/page-collection';
var styles_WizardStepnavItem = [];
export var RenderType_WizardStepnavItem = i0.ɵcrt({ encapsulation: 2,
    styles: styles_WizardStepnavItem, data: {} });
function View_WizardStepnavItem_1(_l) {
    return i0.ɵvid(0, [(_l()(), i0.ɵand(0, null, null, 0))], null, null);
}
export function View_WizardStepnavItem_0(_l) {
    return i0.ɵvid(0, [(_l()(), i0.ɵted(null, ['\n        '])), (_l()(), i0.ɵeld(0, null, null, 4, 'button', [['class', 'btn btn-link clr-wizard-stepnav-link'],
            ['type', 'button']], null, [[null, 'click']], function (_v, en, $event) {
            var ad = true;
            var _co = _v.component;
            if (('click' === en)) {
                var pd_0 = (_co.click() !== false);
                ad = (pd_0 && ad);
            }
            return ad;
        }, null, null)), (_l()(), i0.ɵted(null, ['\n            '])),
        (_l()(), i0.ɵand(16777216, null, null, 1, null, View_WizardStepnavItem_1)),
        i0.ɵdid(540672, null, 0, i2.NgTemplateOutlet, [i0.ViewContainerRef], { ngTemplateOutlet: [0,
                'ngTemplateOutlet'] }, null), (_l()(), i0.ɵted(null, ['\n        '])),
        (_l()(), i0.ɵted(null, ['\n    ']))], function (_ck, _v) {
        var _co = _v.component;
        var currVal_0 = _co.page.navTitle;
        _ck(_v, 4, 0, currVal_0);
    }, null);
}
export function View_WizardStepnavItem_Host_0(_l) {
    return i0.ɵvid(0, [(_l()(), i0.ɵeld(0, null, null, 1, 'div', [['clr-wizard-stepnav-item',
                ''], ['role', 'presentation']], [[8, 'id', 0], [1, 'aria-selected', 0], [1, 'aria-controls',
                0], [2, 'clr-nav-link', null], [2, 'nav-item', null], [2, 'active',
                null], [2, 'disabled', null], [2, 'no-click', null], [2,
                'complete', null]], null, null, View_WizardStepnavItem_0, RenderType_WizardStepnavItem)), i0.ɵdid(49152, null, 0, i1.WizardStepnavItem, [i3.WizardNavigationService, i4.PageCollectionService], null, null)], null, function (_ck, _v) {
        var currVal_0 = i0.ɵnov(_v, 1).id;
        var currVal_1 = i0.ɵnov(_v, 1).isCurrent;
        var currVal_2 = i0.ɵnov(_v, 1).id;
        var currVal_3 = true;
        var currVal_4 = true;
        var currVal_5 = i0.ɵnov(_v, 1).isCurrent;
        var currVal_6 = i0.ɵnov(_v, 1).isDisabled;
        var currVal_7 = !i0.ɵnov(_v, 1).canNavigate;
        var currVal_8 = i0.ɵnov(_v, 1).isComplete;
        _ck(_v, 0, 0, currVal_0, currVal_1, currVal_2, currVal_3, currVal_4, currVal_5, currVal_6, currVal_7, currVal_8);
    });
}
export var WizardStepnavItemNgFactory = i0.ɵccf('[clr-wizard-stepnav-item]', i1.WizardStepnavItem, View_WizardStepnavItem_Host_0, { page: 'page' }, {}, []);
