// clarity-angular/wizard/wizard-page.ngfactory.js
/**
 * @fileoverview This file is generated by the Angular template compiler.
 * Do not edit.
 * @suppress {suspiciousCode,uselessCode,missingProperties,missingOverride}
 */
/* tslint:disable */
import * as i0 from '@angular/core';
import * as i1 from 'clarity-angular/wizard/wizard-page';
import * as i2 from 'clarity-angular/wizard/providers/wizard-navigation';
import * as i3 from 'clarity-angular/wizard/providers/page-collection';
import * as i4 from 'clarity-angular/wizard/providers/button-hub';
var styles_WizardPage = [];
export var RenderType_WizardPage = i0.ɵcrt({ encapsulation: 2, styles: styles_WizardPage,
    data: {} });
export function View_WizardPage_0(_l) {
    return i0.ɵvid(0, [i0.ɵncd(null, 0)], null, null);
}
export function View_WizardPage_Host_0(_l) {
    return i0.ɵvid(0, [(_l()(), i0.ɵeld(0, null, null, 5, 'clr-wizard-page', [['role', 'tabpanel']], [[8, 'id', 0], [1, 'aria-hidden', 0], [1, 'aria-labelledby', 0],
            [2, 'active', null], [2, 'clr-wizard-page', null]], null, null, View_WizardPage_0, RenderType_WizardPage)), i0.ɵdid(114688, null, 4, i1.WizardPage, [i2.WizardNavigationService, i3.PageCollectionService, i4.ButtonHubService], null, null), i0.ɵqud(335544320, 1, { pageTitle: 0 }), i0.ɵqud(335544320, 2, { pageNavTitle: 0 }), i0.ɵqud(335544320, 3, { _buttons: 0 }), i0.ɵqud(335544320, 4, { _headerActions: 0 })], function (_ck, _v) {
        _ck(_v, 1, 0);
    }, function (_ck, _v) {
        var currVal_0 = i0.ɵnov(_v, 1).id;
        var currVal_1 = !i0.ɵnov(_v, 1).current;
        var currVal_2 = i0.ɵnov(_v, 1).stepItemId;
        var currVal_3 = i0.ɵnov(_v, 1).current;
        var currVal_4 = true;
        _ck(_v, 0, 0, currVal_0, currVal_1, currVal_2, currVal_3, currVal_4);
    });
}
export var WizardPageNgFactory = i0.ɵccf('clr-wizard-page', i1.WizardPage, View_WizardPage_Host_0, { nextStepDisabled: 'clrWizardPageNextDisabled',
    previousStepDisabled: 'clrWizardPagePreviousDisabled', preventDefault: 'clrWizardPagePreventDefault',
    stopCancel: 'clrWizardPagePreventDefaultCancel', stopNext: 'clrWizardPagePreventDefaultNext',
    _id: 'id' }, { nextStepDisabledChange: 'clrWizardPageNextDisabledChange', previousStepDisabledChange: 'clrWizardPagePreviousDisabledChange',
    stopCancelChange: 'clrWizardPagePreventDefaultCancelChange', onCommit: 'clrWizardPageOnCommit',
    onLoad: 'clrWizardPageOnLoad', pageOnCancel: 'clrWizardPageOnCancel', finishButtonClicked: 'clrWizardPageFinish',
    previousButtonClicked: 'clrWizardPagePrevious', nextButtonClicked: 'clrWizardPageNext',
    dangerButtonClicked: 'clrWizardPageDanger', primaryButtonClicked: 'clrWizardPagePrimary',
    customButtonClicked: 'clrWizardPageCustomButton' }, ['*']);
