// clarity-angular/wizard/wizard-stepnav.ngfactory.js
/**
 * @fileoverview This file is generated by the Angular template compiler.
 * Do not edit.
 * @suppress {suspiciousCode,uselessCode,missingProperties,missingOverride}
 */
/* tslint:disable */
import * as i0 from '@angular/core';
import * as i1 from './wizard-stepnav-item.ngfactory';
import * as i2 from 'clarity-angular/wizard/wizard-stepnav-item';
import * as i3 from 'clarity-angular/wizard/providers/wizard-navigation';
import * as i4 from 'clarity-angular/wizard/providers/page-collection';
import * as i5 from '@angular/common';
import * as i6 from 'clarity-angular/wizard/wizard-stepnav';
var styles_WizardStepnav = [];
export var RenderType_WizardStepnav = i0.ɵcrt({ encapsulation: 2,
    styles: styles_WizardStepnav, data: {} });
function View_WizardStepnav_1(_l) {
    return i0.ɵvid(0, [(_l()(), i0.ɵeld(0, null, null, 1, 'li', [['class',
                'clr-wizard-stepnav-item'], ['clr-wizard-stepnav-item', ''], ['role', 'presentation']], [[8, 'id', 0], [1, 'aria-selected', 0], [1, 'aria-controls', 0], [2, 'clr-nav-link', null],
            [2, 'nav-item', null], [2, 'active', null], [2, 'disabled', null],
            [2, 'no-click', null], [2, 'complete', null]], null, null, i1.View_WizardStepnavItem_0, i1.RenderType_WizardStepnavItem)),
        i0.ɵdid(49152, null, 0, i2.WizardStepnavItem, [i3.WizardNavigationService,
            i4.PageCollectionService], { page: [0, 'page'] }, null)], function (_ck, _v) {
        var currVal_9 = _v.context.$implicit;
        _ck(_v, 1, 0, currVal_9);
    }, function (_ck, _v) {
        var currVal_0 = i0.ɵnov(_v, 1).id;
        var currVal_1 = i0.ɵnov(_v, 1).isCurrent;
        var currVal_2 = i0.ɵnov(_v, 1).id;
        var currVal_3 = true;
        var currVal_4 = true;
        var currVal_5 = i0.ɵnov(_v, 1).isCurrent;
        var currVal_6 = i0.ɵnov(_v, 1).isDisabled;
        var currVal_7 = !i0.ɵnov(_v, 1).canNavigate;
        var currVal_8 = i0.ɵnov(_v, 1).isComplete;
        _ck(_v, 0, 0, currVal_0, currVal_1, currVal_2, currVal_3, currVal_4, currVal_5, currVal_6, currVal_7, currVal_8);
    });
}
export function View_WizardStepnav_0(_l) {
    return i0.ɵvid(0, [(_l()(), i0.ɵted(null, ['\n        '])), (_l()(), i0.ɵeld(0, null, null, 4, 'ol', [['class', 'clr-wizard-stepnav-list'], ['role',
                'tablist']], null, null, null, null, null)),
        (_l()(), i0.ɵted(null, ['\n            '])), (_l()(), i0.ɵand(16777216, null, null, 1, null, View_WizardStepnav_1)), i0.ɵdid(802816, null, 0, i5.NgForOf, [i0.ViewContainerRef, i0.TemplateRef, i0.IterableDiffers], { ngForOf: [0, 'ngForOf'] }, null), (_l()(), i0.ɵted(null, ['\n        '])),
        (_l()(), i0.ɵted(null, ['\n    ']))], function (_ck, _v) {
        var _co = _v.component;
        var currVal_0 = _co.pageService.pages;
        _ck(_v, 4, 0, currVal_0);
    }, null);
}
export function View_WizardStepnav_Host_0(_l) {
    return i0.ɵvid(0, [(_l()(), i0.ɵeld(0, null, null, 1, 'clr-wizard-stepnav', [['class', 'clr-wizard-stepnav']], null, null, null, View_WizardStepnav_0, RenderType_WizardStepnav)), i0.ɵdid(49152, null, 0, i6.WizardStepnav, [i4.PageCollectionService], null, null)], null, null);
}
export var WizardStepnavNgFactory = i0.ɵccf('clr-wizard-stepnav', i6.WizardStepnav, View_WizardStepnav_Host_0, {}, {}, []);
