// clarity-angular/wizard/wizard-button.ngfactory.js
/**
 * @fileoverview This file is generated by the Angular template compiler.
 * Do not edit.
 * @suppress {suspiciousCode,uselessCode,missingProperties,missingOverride}
 */
/* tslint:disable */
import * as i0 from '@angular/core';
import * as i1 from 'clarity-angular/wizard/wizard-button';
import * as i2 from 'clarity-angular/wizard/providers/wizard-navigation';
import * as i3 from 'clarity-angular/wizard/providers/button-hub';
var styles_WizardButton = ['[aria-hidden="true"][_ngcontent-%COMP%] { display: none; }'];
export var RenderType_WizardButton = i0.ɵcrt({ encapsulation: 0, styles: styles_WizardButton,
    data: {} });
export function View_WizardButton_0(_l) {
    return i0.ɵvid(0, [(_l()(), i0.ɵted(null, ['\n        '])), (_l()(), i0.ɵeld(0, null, null, 3, 'button', [['class', 'btn clr-wizard-btn'], ['type',
                'button']], [[2, 'btn-link', null], [2, 'clr-wizard-btn--tertiary', null],
            [2, 'btn-outline', null], [2, 'clr-wizard-btn--secondary', null],
            [2, 'btn-primary', null], [2, 'clr-wizard-btn--primary', null],
            [2, 'btn-success', null], [2, 'btn-danger', null], [2, 'disabled',
                null]], [[null, 'click']], function (_v, en, $event) {
            var ad = true;
            var _co = _v.component;
            if (('click' === en)) {
                var pd_0 = (_co.click() !== false);
                ad = (pd_0 && ad);
            }
            return ad;
        }, null, null)), (_l()(), i0.ɵted(null, ['\n            '])),
        i0.ɵncd(null, 0), (_l()(), i0.ɵted(null, ['\n        '])), (_l()(),
            i0.ɵted(null, ['\n    ']))], null, function (_ck, _v) {
        var _co = _v.component;
        var currVal_0 = _co.isCancel;
        var currVal_1 = _co.isCancel;
        var currVal_2 = _co.isPrevious;
        var currVal_3 = _co.isPrevious;
        var currVal_4 = _co.isPrimaryAction;
        var currVal_5 = _co.isPrimaryAction;
        var currVal_6 = _co.isFinish;
        var currVal_7 = _co.isDanger;
        var currVal_8 = _co.isDisabled;
        _ck(_v, 1, 0, currVal_0, currVal_1, currVal_2, currVal_3, currVal_4, currVal_5, currVal_6, currVal_7, currVal_8);
    });
}
export function View_WizardButton_Host_0(_l) {
    return i0.ɵvid(0, [(_l()(), i0.ɵeld(0, null, null, 1, 'clr-wizard-button', [['class', 'clr-wizard-btn-wrapper']], [[1, 'aria-hidden', 0]], null, null, View_WizardButton_0, RenderType_WizardButton)), i0.ɵdid(49152, null, 0, i1.WizardButton, [i2.WizardNavigationService, i3.ButtonHubService], null, null)], null, function (_ck, _v) {
        var currVal_0 = i0.ɵnov(_v, 1).isHidden;
        _ck(_v, 0, 0, currVal_0);
    });
}
export var WizardButtonNgFactory = i0.ɵccf('clr-wizard-button', i1.WizardButton, View_WizardButton_Host_0, { type: 'type', disabled: 'clrWizardButtonDisabled',
    hidden: 'clrWizardButtonHidden' }, { wasClicked: 'clrWizardButtonClicked' }, ['*']);
