// clarity-angular/button/button-group/button-group.ngfactory.js
/**
 * @fileoverview This file is generated by the Angular template compiler.
 * Do not edit.
 * @suppress {suspiciousCode,uselessCode,missingProperties,missingOverride}
 */
/* tslint:disable */
import * as i0 from '@angular/core';
import * as i1 from '@angular/common';
import * as i2 from 'clarity-angular/icon/icon';
import * as i3 from 'clarity-angular/popover/common/popover-old.directive';
import * as i4 from 'clarity-angular/button/button-group/button-group';
import * as i5 from 'clarity-angular/button/providers/buttonInGroup.service';
var styles_ButtonGroup = [];
export var RenderType_ButtonGroup = i0.ɵcrt({ encapsulation: 2, styles: styles_ButtonGroup,
    data: {} });
function View_ButtonGroup_2(_l) {
    return i0.ɵvid(0, [(_l()(), i0.ɵand(0, null, null, 0))], null, null);
}
function View_ButtonGroup_1(_l) {
    return i0.ɵvid(0, [(_l()(), i0.ɵeld(0, null, null, 4, null, null, null, null, null, null, null)),
        (_l()(), i0.ɵted(null, ['\n          '])), (_l()(), i0.ɵand(16777216, null, null, 1, null, View_ButtonGroup_2)), i0.ɵdid(540672, null, 0, i1.NgTemplateOutlet, [i0.ViewContainerRef], { ngTemplateOutlet: [0, 'ngTemplateOutlet'] }, null), (_l()(), i0.ɵted(null, ['\n      ']))], function (_ck, _v) {
        var currVal_0 = _v.context.$implicit.templateRef;
        _ck(_v, 3, 0, currVal_0);
    }, null);
}
function View_ButtonGroup_5(_l) {
    return i0.ɵvid(0, [(_l()(), i0.ɵand(0, null, null, 0))], null, null);
}
function View_ButtonGroup_4(_l) {
    return i0.ɵvid(0, [(_l()(), i0.ɵeld(0, null, null, 4, 'div', [['class',
                'dropdown-menu']], null, null, null, null, null)),
        (_l()(), i0.ɵted(null, ['\n                  '])), (_l()(), i0.ɵand(16777216, null, null, 1, null, View_ButtonGroup_5)), i0.ɵdid(540672, null, 0, i1.NgTemplateOutlet, [i0.ViewContainerRef], { ngTemplateOutlet: [0,
                'ngTemplateOutlet'] }, null), (_l()(), i0.ɵted(null, ['\n              ']))], function (_ck, _v) {
        var currVal_0 = i0.ɵnov(_v.parent.parent, 7);
        _ck(_v, 3, 0, currVal_0);
    }, null);
}
function View_ButtonGroup_3(_l) {
    return i0.ɵvid(0, [(_l()(), i0.ɵeld(0, null, null, 14, null, null, null, null, null, null, null)),
        (_l()(), i0.ɵted(null, ['\n          '])), (_l()(), i0.ɵeld(0, [['anchor',
                1]], null, 11, 'div', [['class', 'btn-group-overflow open']], null, null, null, null, null)), i0.ɵdid(278528, null, 0, i1.NgClass, [i0.IterableDiffers, i0.KeyValueDiffers, i0.ElementRef,
            i0.Renderer], { klass: [0, 'klass'], ngClass: [1, 'ngClass'] }, null),
        (_l()(), i0.ɵted(null, ['\n              '])), (_l()(), i0.ɵeld(0, null, null, 4, 'button', [['class', 'btn dropdown-toggle']], null, [[null, 'click']], function (_v, en, $event) {
            var ad = true;
            var _co = _v.component;
            if (('click' === en)) {
                var pd_0 = (_co.toggleMenu() !== false);
                ad = (pd_0 && ad);
            }
            return ad;
        }, null, null)), (_l()(), i0.ɵted(null, ['\n                  '])),
        (_l()(), i0.ɵeld(0, null, null, 1, 'clr-icon', [['shape', 'ellipsis-horizontal']], null, null, null, null, null)),
        i0.ɵdid(16384, null, 0, i2.IconCustomTag, [], null, null),
        (_l()(), i0.ɵted(null, ['\n              '])), (_l()(), i0.ɵted(null, ['\n              '])), (_l()(), i0.ɵand(16777216, null, null, 1, null, View_ButtonGroup_4)), i0.ɵdid(147456, null, 0, i3.PopoverDirectiveOld, [i0.TemplateRef, i0.ViewContainerRef], { anchorElem: [0, 'anchorElem'], anchorPoint: [1,
                'anchorPoint'], popoverPoint: [2, 'popoverPoint'], clrPopoverOld: [3, 'clrPopoverOld'] }, null), (_l()(), i0.ɵted(null, ['\n          '])), (_l()(),
            i0.ɵted(null, ['\n      ']))], function (_ck, _v) {
        var _co = _v.component;
        var currVal_0 = 'btn-group-overflow open';
        var currVal_1 = _co.menuPosition;
        _ck(_v, 3, 0, currVal_0, currVal_1);
        var currVal_2 = i0.ɵnov(_v, 2);
        var currVal_3 = _co.anchorPoint;
        var currVal_4 = _co.popoverPoint;
        var currVal_5 = _co.openMenu;
        _ck(_v, 12, 0, currVal_2, currVal_3, currVal_4, currVal_5);
    }, null);
}
function View_ButtonGroup_8(_l) {
    return i0.ɵvid(0, [(_l()(), i0.ɵand(0, null, null, 0))], null, null);
}
function View_ButtonGroup_7(_l) {
    return i0.ɵvid(0, [(_l()(), i0.ɵeld(0, null, null, 4, null, null, null, null, null, null, null)),
        (_l()(), i0.ɵted(null, ['\n              '])), (_l()(), i0.ɵand(16777216, null, null, 1, null, View_ButtonGroup_8)), i0.ɵdid(540672, null, 0, i1.NgTemplateOutlet, [i0.ViewContainerRef], { ngTemplateOutlet: [0,
                'ngTemplateOutlet'] }, null), (_l()(), i0.ɵted(null, ['\n          ']))], function (_ck, _v) {
        var currVal_0 = _v.context.$implicit.templateRef;
        _ck(_v, 3, 0, currVal_0);
    }, null);
}
function View_ButtonGroup_6(_l) {
    return i0.ɵvid(0, [(_l()(), i0.ɵted(null, ['\n          '])), (_l()(), i0.ɵand(16777216, null, null, 1, null, View_ButtonGroup_7)), i0.ɵdid(802816, null, 0, i1.NgForOf, [i0.ViewContainerRef, i0.TemplateRef, i0.IterableDiffers], { ngForOf: [0, 'ngForOf'] }, null), (_l()(), i0.ɵted(null, ['\n      ']))], function (_ck, _v) {
        var _co = _v.component;
        var currVal_0 = _co.menuButtons;
        _ck(_v, 2, 0, currVal_0);
    }, null);
}
export function View_ButtonGroup_0(_l) {
    return i0.ɵvid(0, [(_l()(), i0.ɵted(null, ['\n      '])), (_l()(), i0.ɵand(16777216, null, null, 1, null, View_ButtonGroup_1)), i0.ɵdid(802816, null, 0, i1.NgForOf, [i0.ViewContainerRef, i0.TemplateRef, i0.IterableDiffers], { ngForOf: [0, 'ngForOf'] }, null), (_l()(), i0.ɵted(null, ['\n      '])),
        (_l()(), i0.ɵand(16777216, null, null, 1, null, View_ButtonGroup_3)),
        i0.ɵdid(16384, null, 0, i1.NgIf, [i0.ViewContainerRef, i0.TemplateRef], { ngIf: [0,
                'ngIf'] }, null), (_l()(), i0.ɵted(null, ['\n      '])), (_l()(),
            i0.ɵand(0, [['ref', 2]], null, 0, null, View_ButtonGroup_6)),
        (_l()(), i0.ɵted(null, ['\n    ']))], function (_ck, _v) {
        var _co = _v.component;
        var currVal_0 = _co.inlineButtons;
        _ck(_v, 2, 0, currVal_0);
        var currVal_1 = (_co.menuButtons.length > 0);
        _ck(_v, 5, 0, currVal_1);
    }, null);
}
export function View_ButtonGroup_Host_0(_l) {
    return i0.ɵvid(0, [(_l()(), i0.ɵeld(0, null, null, 3, 'clr-button-group', [], [[2, 'btn-group', null]], [['document', 'click']], function (_v, en, $event) {
            var ad = true;
            if (('document:click' === en)) {
                var pd_0 = (i0.ɵnov(_v, 2).onMouseClick($event.target) !== false);
                ad = (pd_0 && ad);
            }
            return ad;
        }, View_ButtonGroup_0, RenderType_ButtonGroup)), i0.ɵprd(512, null, i5.ButtonInGroupService, i5.ButtonInGroupService, []), i0.ɵdid(1097728, null, 1, i4.ButtonGroup, [i5.ButtonInGroupService, i0.ElementRef], null, null), i0.ɵqud(603979776, 1, { buttons: 1 })], null, function (_ck, _v) {
        var currVal_0 = true;
        _ck(_v, 0, 0, currVal_0);
    });
}
export var ButtonGroupNgFactory = i0.ɵccf('clr-button-group', i4.ButtonGroup, View_ButtonGroup_Host_0, { menuPosition: 'clrMenuPosition' }, {}, []);
