// clarity-angular/button/button-group/button.ngfactory.js
/**
 * @fileoverview This file is generated by the Angular template compiler.
 * Do not edit.
 * @suppress {suspiciousCode,uselessCode,missingProperties,missingOverride}
 */
/* tslint:disable */
import * as i0 from '@angular/core';
import * as i1 from 'clarity-angular/button/button-group/button';
import * as i2 from 'clarity-angular/button/providers/buttonInGroup.service';
var styles_Button = [];
export var RenderType_Button = i0.ɵcrt({ encapsulation: 2, styles: styles_Button,
    data: {} });
function View_Button_1(_l) {
    return i0.ɵvid(0, [(_l()(), i0.ɵted(null, ['\n            '])), (_l()(), i0.ɵeld(0, null, null, 3, 'button', [], [[8, 'className', 0], [1, 'type',
                0], [1, 'name', 0], [1, 'disabled', 0]], [[null, 'click']], function (_v, en, $event) {
            var ad = true;
            var _co = _v.component;
            if (('click' === en)) {
                var pd_0 = (_co.emitClick() !== false);
                ad = (pd_0 && ad);
            }
            return ad;
        }, null, null)), (_l()(), i0.ɵted(null, ['\n                '])),
        i0.ɵncd(null, 0), (_l()(), i0.ɵted(null, ['\n            '])),
        (_l()(), i0.ɵted(null, ['\n        ']))], null, function (_ck, _v) {
        var _co = _v.component;
        var currVal_0 = _co.classNames;
        var currVal_1 = _co.type;
        var currVal_2 = _co.name;
        var currVal_3 = _co.disabled;
        _ck(_v, 1, 0, currVal_0, currVal_1, currVal_2, currVal_3);
    });
}
export function View_Button_0(_l) {
    return i0.ɵvid(0, [i0.ɵqud(402653184, 1, { templateRef: 0 }), (_l()(), i0.ɵted(null, ['\n        '])), (_l()(), i0.ɵand(0, [[1, 2], ['buttonProjectedRef', 2]], null, 0, null, View_Button_1)), (_l()(), i0.ɵted(null, ['\n    ']))], null, null);
}
export function View_Button_Host_0(_l) {
    return i0.ɵvid(0, [(_l()(), i0.ɵeld(0, null, null, 1, 'clr-button', [], null, null, null, View_Button_0, RenderType_Button)),
        i0.ɵdid(4243456, null, 0, i1.Button, [[3, i2.ButtonInGroupService]], null, null)], null, null);
}
export var ButtonNgFactory = i0.ɵccf('clr-button', i1.Button, View_Button_Host_0, { inMenu: 'clrInMenu', classNames: 'class', name: 'name',
    type: 'type', disabled: 'disabled' }, { _click: 'click' }, ['*']);
