// clarity-angular/emphasis/alert/alert.ngfactory.js
/**
 * @fileoverview This file is generated by the Angular template compiler.
 * Do not edit.
 * @suppress {suspiciousCode,uselessCode,missingProperties,missingOverride}
 */
/* tslint:disable */
import * as i0 from '@angular/core';
import * as i1 from 'clarity-angular/icon/icon';
import * as i2 from '@angular/common';
import * as i3 from 'clarity-angular/emphasis/alert/alert';
import * as i4 from 'clarity-angular/emphasis/alert/providers/icon-and-types-service';
var styles_Alert = [];
export var RenderType_Alert = i0.ɵcrt({ encapsulation: 2, styles: styles_Alert,
    data: {} });
function View_Alert_2(_l) {
    return i0.ɵvid(0, [(_l()(), i0.ɵeld(0, null, null, 4, 'button', [['aria-label',
                'Close'], ['class', 'close'], ['type', 'button']], null, [[null,
                'click']], function (_v, en, $event) {
            var ad = true;
            var _co = _v.component;
            if (('click' === en)) {
                var pd_0 = (_co.close() !== false);
                ad = (pd_0 && ad);
            }
            return ad;
        }, null, null)), (_l()(), i0.ɵted(null, ['\n          '])),
        (_l()(), i0.ɵeld(0, null, null, 1, 'clr-icon', [['aria-hidden', 'true'],
            ['shape', 'close']], null, null, null, null, null)), i0.ɵdid(16384, null, 0, i1.IconCustomTag, [], null, null), (_l()(), i0.ɵted(null, ['\n      ']))], null, null);
}
function View_Alert_1(_l) {
    return i0.ɵvid(0, [(_l()(), i0.ɵeld(0, null, null, 10, 'div', [['class',
                'alert']], [[2, 'alert-sm', null], [2, 'alert-app-level', null]], null, null, null, null)), i0.ɵdid(278528, null, 0, i2.NgClass, [i0.IterableDiffers, i0.KeyValueDiffers, i0.ElementRef, i0.Renderer], { klass: [0, 'klass'], ngClass: [1, 'ngClass'] }, null), (_l()(), i0.ɵted(null, ['\n      '])), (_l()(), i0.ɵeld(0, null, null, 3, 'div', [['class',
                'alert-items']], null, null, null, null, null)),
        (_l()(), i0.ɵted(null, ['\n          '])), i0.ɵncd(null, 0), (_l()(),
            i0.ɵted(null, ['\n      '])), (_l()(), i0.ɵted(null, ['\n      '])),
        (_l()(), i0.ɵand(16777216, null, null, 1, null, View_Alert_2)),
        i0.ɵdid(16384, null, 0, i2.NgIf, [i0.ViewContainerRef, i0.TemplateRef], { ngIf: [0,
                'ngIf'] }, null), (_l()(), i0.ɵted(null, ['\n  ']))], function (_ck, _v) {
        var _co = _v.component;
        var currVal_2 = 'alert';
        var currVal_3 = _co.alertClass;
        _ck(_v, 1, 0, currVal_2, currVal_3);
        var currVal_4 = _co.closable;
        _ck(_v, 9, 0, currVal_4);
    }, function (_ck, _v) {
        var _co = _v.component;
        var currVal_0 = _co.isSmall;
        var currVal_1 = _co.isAppLevel;
        _ck(_v, 0, 0, currVal_0, currVal_1);
    });
}
export function View_Alert_0(_l) {
    return i0.ɵvid(0, [(_l()(), i0.ɵted(null, ['\n  '])), (_l()(), i0.ɵted(null, ['\n\n  '])), (_l()(), i0.ɵand(16777216, null, null, 1, null, View_Alert_1)), i0.ɵdid(16384, null, 0, i2.NgIf, [i0.ViewContainerRef, i0.TemplateRef], { ngIf: [0, 'ngIf'] }, null), (_l()(), i0.ɵted(null, ['\n']))], function (_ck, _v) {
        var _co = _v.component;
        var currVal_0 = !_co._closed;
        _ck(_v, 3, 0, currVal_0);
    }, null);
}
export function View_Alert_Host_0(_l) {
    return i0.ɵvid(0, [(_l()(), i0.ɵeld(0, null, null, 2, 'clr-alert', [], null, null, null, View_Alert_0, RenderType_Alert)), i0.ɵprd(512, null, i4.AlertIconAndTypesService, i4.AlertIconAndTypesService, []),
        i0.ɵdid(49152, null, 0, i3.Alert, [i4.AlertIconAndTypesService], null, null)], null, null);
}
export var AlertNgFactory = i0.ɵccf('clr-alert', i3.Alert, View_Alert_Host_0, { isSmall: 'clrAlertSizeSmall', closable: 'clrAlertClosable', isAppLevel: 'clrAlertAppLevel',
    _closed: 'clrAlertClosed', alertType: 'clrAlertType', alertIconShape: 'clrAlertIcon' }, { _closedChanged: 'clrAlertClosedChange' }, ['*']);
