// clarity-angular/layout/tabs/tabs.ngfactory.js
/**
 * @fileoverview This file is generated by the Angular template compiler.
 * Do not edit.
 * @suppress {suspiciousCode,uselessCode,missingProperties,missingOverride}
 */
/* tslint:disable */
import * as i0 from '@angular/core';
import * as i1 from '@angular/common';
import * as i2 from 'clarity-angular/icon/icon';
import * as i3 from './tab-overflow-content.ngfactory';
import * as i4 from 'clarity-angular/layout/tabs/tab-overflow-content';
import * as i5 from 'clarity-angular/layout/tabs/tabs';
import * as i6 from 'clarity-angular/utils/conditional/if-active.service';
import * as i7 from 'clarity-angular/utils/conditional/if-open.service';
import * as i8 from 'clarity-angular/layout/tabs/tabs-service';
var styles_Tabs = [];
export var RenderType_Tabs = i0.ɵcrt({ encapsulation: 2, styles: styles_Tabs,
    data: {} });
function View_Tabs_2(_l) {
    return i0.ɵvid(0, [(_l()(), i0.ɵeld(16777216, null, null, 2, null, null, null, null, null, null, null)),
        i0.ɵdid(540672, null, 0, i1.NgTemplateOutlet, [i0.ViewContainerRef], { ngTemplateOutlet: [0,
                'ngTemplateOutlet'] }, null), (_l()(), i0.ɵted(null, ['\n                ']))], function (_ck, _v) {
        var currVal_0 = _v.parent.context.$implicit.templateRefContainer.template;
        _ck(_v, 1, 0, currVal_0);
    }, null);
}
function View_Tabs_1(_l) {
    return i0.ɵvid(0, [(_l()(), i0.ɵeld(0, null, null, 4, null, null, null, null, null, null, null)),
        (_l()(), i0.ɵted(null, ['\n                '])), (_l()(), i0.ɵand(16777216, null, null, 1, null, View_Tabs_2)), i0.ɵdid(16384, null, 0, i1.NgIf, [i0.ViewContainerRef, i0.TemplateRef], { ngIf: [0, 'ngIf'] }, null), (_l()(), i0.ɵted(null, ['\n            ']))], function (_ck, _v) {
        var currVal_0 = !_v.context.$implicit.inOverflow;
        _ck(_v, 3, 0, currVal_0);
    }, null);
}
function View_Tabs_5(_l) {
    return i0.ɵvid(0, [(_l()(), i0.ɵeld(16777216, null, null, 2, null, null, null, null, null, null, null)),
        i0.ɵdid(540672, null, 0, i1.NgTemplateOutlet, [i0.ViewContainerRef], { ngTemplateOutlet: [0,
                'ngTemplateOutlet'] }, null), (_l()(), i0.ɵted(null, ['\n                            ']))], function (_ck, _v) {
        var currVal_0 = _v.parent.context.$implicit.templateRefContainer.template;
        _ck(_v, 1, 0, currVal_0);
    }, null);
}
function View_Tabs_4(_l) {
    return i0.ɵvid(0, [(_l()(), i0.ɵeld(0, null, null, 4, null, null, null, null, null, null, null)),
        (_l()(), i0.ɵted(null, ['\n                            '])), (_l()(), i0.ɵand(16777216, null, null, 1, null, View_Tabs_5)), i0.ɵdid(16384, null, 0, i1.NgIf, [i0.ViewContainerRef, i0.TemplateRef], { ngIf: [0, 'ngIf'] }, null), (_l()(), i0.ɵted(null, ['\n                        ']))], function (_ck, _v) {
        var currVal_0 = _v.context.$implicit.inOverflow;
        _ck(_v, 3, 0, currVal_0);
    }, null);
}
function View_Tabs_3(_l) {
    return i0.ɵvid(0, [(_l()(), i0.ɵeld(0, null, null, 21, null, null, null, null, null, null, null)),
        (_l()(), i0.ɵted(null, ['\n                '])), (_l()(), i0.ɵeld(0, null, null, 18, 'div', [['class', 'tabs-overflow bottom-right']], [[2, 'open',
                null]], [[null, 'click']], function (_v, en, $event) {
            var ad = true;
            var _co = _v.component;
            if (('click' === en)) {
                var pd_0 = (_co.toggleOverflow($event) !== false);
                ad = (pd_0 && ad);
            }
            return ad;
        }, null, null)), (_l()(), i0.ɵted(null, ['\n                    '])),
        (_l()(), i0.ɵeld(0, null, null, 7, 'li', [['class', 'nav-item'], ['role',
                'presentation']], null, null, null, null, null)), (_l()(), i0.ɵted(null, ['\n                        '])),
        (_l()(), i0.ɵeld(0, null, null, 4, 'button', [['class', 'btn btn-link nav-link dropdown-toggle']], [[2, 'active', null]], null, null, null, null)),
        (_l()(), i0.ɵted(null, ['\n                            '])), (_l()(), i0.ɵeld(0, null, null, 1, 'clr-icon', [['shape', 'ellipsis-horizontal']], [[2, 'is-info', null]], null, null, null, null)), i0.ɵdid(16384, null, 0, i2.IconCustomTag, [], null, null), (_l()(), i0.ɵted(null, ['\n                        '])),
        (_l()(), i0.ɵted(null, ['\n                    '])), (_l()(), i0.ɵted(null, ['\n                    '])), (_l()(), i0.ɵted(null, ['\n                    '])),
        (_l()(), i0.ɵeld(0, null, null, 5, 'clr-tab-overflow-content', [], [[2, 'dropdown-menu', null], [4, 'visibility', null]], null, null, i3.View_TabOverflowContent_0, i3.RenderType_TabOverflowContent)),
        i0.ɵdid(8568832, null, 0, i4.TabOverflowContent, [i0.Injector, [1, i0.ElementRef]], null, null), (_l()(), i0.ɵted(0, ['\n                        '])),
        (_l()(), i0.ɵand(16777216, null, 0, 1, null, View_Tabs_4)), i0.ɵdid(802816, null, 0, i1.NgForOf, [i0.ViewContainerRef, i0.TemplateRef, i0.IterableDiffers], { ngForOf: [0, 'ngForOf'] }, null), (_l()(), i0.ɵted(0, ['\n                    '])),
        (_l()(), i0.ɵted(null, ['\n                '])), (_l()(), i0.ɵted(null, ['\n            ']))], function (_ck, _v) {
        var _co = _v.component;
        var currVal_5 = _co.tabLinkDirectives;
        _ck(_v, 18, 0, currVal_5);
    }, function (_ck, _v) {
        var _co = _v.component;
        var currVal_0 = _co.ifOpenService.open;
        _ck(_v, 2, 0, currVal_0);
        var currVal_1 = _co.activeTabInOverflow;
        _ck(_v, 6, 0, currVal_1);
        var currVal_2 = _co.ifOpenService.open;
        _ck(_v, 8, 0, currVal_2);
        var currVal_3 = true;
        var currVal_4 = i0.ɵnov(_v, 15).hidden;
        _ck(_v, 14, 0, currVal_3, currVal_4);
    });
}
export function View_Tabs_0(_l) {
    return i0.ɵvid(0, [(_l()(), i0.ɵted(null, ['        \n        '])), (_l()(),
            i0.ɵeld(0, null, null, 8, 'ul', [['class', 'nav'], ['role', 'tablist']], null, null, null, null, null)),
        (_l()(), i0.ɵted(null, ['\n            '])), (_l()(), i0.ɵted(null, ['\n            '])), (_l()(), i0.ɵand(16777216, null, null, 1, null, View_Tabs_1)), i0.ɵdid(802816, null, 0, i1.NgForOf, [i0.ViewContainerRef, i0.TemplateRef, i0.IterableDiffers], { ngForOf: [0, 'ngForOf'] }, null), (_l()(), i0.ɵted(null, ['\n            '])), (_l()(),
            i0.ɵand(16777216, null, null, 1, null, View_Tabs_3)),
        i0.ɵdid(16384, null, 0, i1.NgIf, [i0.ViewContainerRef, i0.TemplateRef], { ngIf: [0,
                'ngIf'] }, null), (_l()(), i0.ɵted(null, ['\n        '])), (_l()(),
            i0.ɵted(null, ['\n        '])), (_l()(), i0.ɵted(null, ['\n        '])),
        i0.ɵncd(null, 0), (_l()(), i0.ɵted(null, ['\n    ']))], function (_ck, _v) {
        var _co = _v.component;
        var currVal_0 = _co.tabLinkDirectives;
        _ck(_v, 5, 0, currVal_0);
        var currVal_1 = (_co.tabsService.overflowTabs.length > 0);
        _ck(_v, 8, 0, currVal_1);
    }, null);
}
export function View_Tabs_Host_0(_l) {
    return i0.ɵvid(0, [(_l()(), i0.ɵeld(0, null, null, 5, 'clr-tabs', [], null, null, null, View_Tabs_0, RenderType_Tabs)), i0.ɵprd(512, null, i6.IfActiveService, i6.IfActiveService, []), i0.ɵprd(512, null, i7.IfOpenService, i7.IfOpenService, []), i0.ɵprd(512, null, i8.TabsService, i8.TabsService, []), i0.ɵdid(49152, null, 1, i5.Tabs, [i6.IfActiveService, i7.IfOpenService, i8.TabsService], null, null),
        i0.ɵqud(603979776, 1, { tabLinkDirectives: 1 })], null, null);
}
export var TabsNgFactory = i0.ɵccf('clr-tabs', i5.Tabs, View_Tabs_Host_0, {}, {}, ['*']);
