// clarity-angular/layout/nav/header.ngfactory.js
/**
 * @fileoverview This file is generated by the Angular template compiler.
 * Do not edit.
 * @suppress {suspiciousCode,uselessCode,missingProperties,missingOverride}
 */
/* tslint:disable */
import * as i0 from '@angular/core';
import * as i1 from '@angular/common';
import * as i2 from 'clarity-angular/layout/nav/header';
import * as i3 from 'clarity-angular/layout/nav/clrResponsiveNavigationService';
var styles_Header = [];
export var RenderType_Header = i0.ɵcrt({ encapsulation: 2, styles: styles_Header,
    data: {} });
function View_Header_1(_l) {
    return i0.ɵvid(0, [(_l()(), i0.ɵeld(0, null, null, 3, 'button', [['class',
                'header-hamburger-trigger'], ['type', 'button']], null, [[null,
                'click']], function (_v, en, $event) {
            var ad = true;
            var _co = _v.component;
            if (('click' === en)) {
                var pd_0 = (_co.toggleNav(_co.responsiveNavCodes.NAV_LEVEL_1) !== false);
                ad = (pd_0 && ad);
            }
            return ad;
        }, null, null)), (_l()(), i0.ɵted(null, ['\n            '])),
        (_l()(), i0.ɵeld(0, null, null, 0, 'span', [], null, null, null, null, null)), (_l()(), i0.ɵted(null, ['\n        ']))], null, null);
}
function View_Header_2(_l) {
    return i0.ɵvid(0, [(_l()(), i0.ɵeld(0, null, null, 3, 'button', [['class',
                'header-overflow-trigger'], ['type', 'button']], null, [[null,
                'click']], function (_v, en, $event) {
            var ad = true;
            var _co = _v.component;
            if (('click' === en)) {
                var pd_0 = (_co.toggleNav(_co.responsiveNavCodes.NAV_LEVEL_2) !== false);
                ad = (pd_0 && ad);
            }
            return ad;
        }, null, null)), (_l()(), i0.ɵted(null, ['\n            '])),
        (_l()(), i0.ɵeld(0, null, null, 0, 'span', [], null, null, null, null, null)), (_l()(), i0.ɵted(null, ['\n        ']))], null, null);
}
export function View_Header_0(_l) {
    return i0.ɵvid(0, [(_l()(), i0.ɵted(null, ['\n        '])), (_l()(), i0.ɵand(16777216, null, null, 1, null, View_Header_1)), i0.ɵdid(16384, null, 0, i1.NgIf, [i0.ViewContainerRef, i0.TemplateRef], { ngIf: [0, 'ngIf'] }, null),
        (_l()(), i0.ɵted(null, ['\n        '])), i0.ɵncd(null, 0), (_l()(),
            i0.ɵted(null, ['\n        '])), (_l()(), i0.ɵand(16777216, null, null, 1, null, View_Header_2)), i0.ɵdid(16384, null, 0, i1.NgIf, [i0.ViewContainerRef, i0.TemplateRef], { ngIf: [0, 'ngIf'] }, null),
        (_l()(), i0.ɵted(null, ['\n        '])), (_l()(), i0.ɵeld(0, null, null, 0, 'div', [['class', 'header-backdrop']], null, [[null,
                'click']], function (_v, en, $event) {
            var ad = true;
            var _co = _v.component;
            if (('click' === en)) {
                var pd_0 = (_co.closeOpenNav() !== false);
                ad = (pd_0 && ad);
            }
            return ad;
        }, null, null)), (_l()(), i0.ɵted(null, ['\n    ']))], function (_ck, _v) {
        var _co = _v.component;
        var currVal_0 = _co.isNavLevel1OnPage;
        _ck(_v, 2, 0, currVal_0);
        var currVal_1 = _co.isNavLevel2OnPage;
        _ck(_v, 7, 0, currVal_1);
    }, null);
}
export function View_Header_Host_0(_l) {
    return i0.ɵvid(0, [(_l()(), i0.ɵeld(0, null, null, 1, 'clr-header', [], [[2, 'header', null]], null, null, View_Header_0, RenderType_Header)),
        i0.ɵdid(245760, null, 0, i2.Header, [i3.ClrResponsiveNavigationService], null, null)], function (_ck, _v) {
        _ck(_v, 1, 0);
    }, function (_ck, _v) {
        var currVal_0 = true;
        _ck(_v, 0, 0, currVal_0);
    });
}
export var HeaderNgFactory = i0.ɵccf('clr-header', i2.Header, View_Header_Host_0, {}, {}, ['*']);
