// clarity-angular/modal/modal.ngfactory.js
/**
 * @fileoverview This file is generated by the Angular template compiler.
 * Do not edit.
 * @suppress {suspiciousCode,uselessCode,missingProperties,missingOverride}
 */
/* tslint:disable */
import * as i0 from '@angular/core';
import * as i1 from 'clarity-angular/icon/icon';
import * as i2 from 'clarity-angular/utils/focus-trap/focus-trap.directive';
import * as i3 from '@angular/common';
import * as i4 from 'clarity-angular/modal/modal';
import * as i5 from 'clarity-angular/utils/scrolling/scrolling-service';
import * as i6 from '@angular/platform-browser';
var styles_Modal = ['[_nghost-%COMP%] { display: inline-block; }'];
export var RenderType_Modal = i0.ɵcrt({ encapsulation: 0, styles: styles_Modal,
    data: { 'animation': [{ type: 7, name: 'fadeDown', definitions: [{ type: 1, expr: '* => false',
                        animation: [{ type: 6, styles: { opacity: 0, transform: 'translate(0, -25%)' }, offset: null },
                            { type: 4, styles: null, timings: '0.2s ease-in-out' }], options: null },
                    { type: 1, expr: 'false => *', animation: [{ type: 4, styles: { type: 6, styles: { opacity: 0,
                                        transform: 'translate(0, -25%)' }, offset: null }, timings: '0.2s ease-in-out' }],
                        options: null }], options: {} }, { type: 7, name: 'fade', definitions: [{ type: 1,
                        expr: 'void => *', animation: [{ type: 6, styles: { opacity: 0 }, offset: null },
                            { type: 4, styles: { type: 6, styles: { opacity: 0.85 }, offset: null }, timings: '0.2s ease-in-out' }],
                        options: null }, { type: 1, expr: '* => void', animation: [{ type: 4, styles: { type: 6,
                                    styles: { opacity: 0 }, offset: null }, timings: '0.2s ease-in-out' }], options: null }],
                options: {} }, { type: 7, name: 'ghostPageOneState', definitions: [{ type: 0, name: 'inactive',
                        styles: { type: 6, styles: { left: '-24px' }, offset: null }, options: undefined },
                    { type: 0, name: 'ready', styles: { type: 6, styles: { left: '0' }, offset: null },
                        options: undefined }, { type: 0, name: 'penultimateGhost', styles: { type: 6,
                            styles: { left: '-24px' }, offset: null }, options: undefined },
                    { type: 0, name: 'lastGhost', styles: { type: 6, styles: { left: '-24px' }, offset: null },
                        options: undefined }, { type: 1, expr: 'inactive => *', animation: { type: 4,
                            styles: null, timings: '100ms ease-out' }, options: null },
                    { type: 1, expr: 'ready => *', animation: { type: 4, styles: null, timings: '100ms ease-in' },
                        options: null }, { type: 1, expr: 'lastGhost => *', animation: { type: 4,
                            styles: null, timings: '100ms ease-out' }, options: null },
                    { type: 1, expr: 'penultimateGhost => *', animation: { type: 4, styles: null,
                            timings: '100ms ease-in' }, options: null }], options: {} }, { type: 7, name: 'ghostPageTwoState',
                definitions: [{ type: 0, name: 'inactive', styles: { type: 6, styles: { left: '-24px', top: '24px',
                                bottom: '24px' }, offset: null }, options: undefined }, { type: 0,
                        name: 'ready', styles: { type: 6, styles: { left: '24px' }, offset: null },
                        options: undefined }, { type: 0, name: 'penultimateGhost', styles: { type: 6,
                            styles: { left: '0px', top: '24px', bottom: '24px', background: '#bbb' }, offset: null },
                        options: undefined }, { type: 0, name: 'lastGhost', styles: { type: 6, styles: { left: '-24px',
                                top: '24px', bottom: '24px' }, offset: null }, options: undefined },
                    { type: 1, expr: 'inactive => *', animation: { type: 4, styles: null, timings: '100ms ease-out' },
                        options: null }, { type: 1, expr: 'ready => *', animation: { type: 4,
                            styles: null, timings: '100ms ease-in' }, options: null },
                    { type: 1, expr: 'lastGhost => *', animation: { type: 4, styles: null, timings: '100ms ease-out' },
                        options: null }, { type: 1, expr: 'penultimateGhost => *', animation: { type: 4,
                            styles: null, timings: '100ms ease-in' }, options: null }],
                options: {} }] } });
function View_Modal_2(_l) {
    return i0.ɵvid(0, [(_l()(), i0.ɵeld(0, null, null, 4, 'button', [['aria-label',
                'Close'], ['class', 'close'], ['type', 'button']], null, [[null,
                'click']], function (_v, en, $event) {
            var ad = true;
            var _co = _v.component;
            if (('click' === en)) {
                var pd_0 = (_co.close() !== false);
                ad = (pd_0 && ad);
            }
            return ad;
        }, null, null)), (_l()(), i0.ɵted(null, ['\n                                  '])),
        (_l()(), i0.ɵeld(0, null, null, 1, 'clr-icon', [['aria-hidden', 'true'],
            ['shape', 'close']], null, null, null, null, null)), i0.ɵdid(16384, null, 0, i1.IconCustomTag, [], null, null), (_l()(), i0.ɵted(null, ['\n                              ']))], null, null);
}
function View_Modal_1(_l) {
    return i0.ɵvid(0, [(_l()(), i0.ɵeld(0, null, null, 39, 'div', [['class',
                'modal'], ['clrFocusTrap', '']], null, [['document', 'focusin']], function (_v, en, $event) {
            var ad = true;
            if (('document:focusin' === en)) {
                var pd_0 = (i0.ɵnov(_v, 1).onFocusIn($event) !== false);
                ad = (pd_0 && ad);
            }
            return ad;
        }, null, null)), i0.ɵdid(4210688, null, 0, i2.FocusTrapDirective, [i0.ElementRef], null, null), (_l()(), i0.ɵted(null, ['\n          '])),
        (_l()(), i0.ɵted(null, ['\n          '])), (_l()(), i0.ɵeld(0, null, null, 32, 'div', [['aria-hidden', 'true'], ['class', 'modal-dialog'],
            ['role', 'dialog']], [[24, '@fadeDown', 0], [2, 'modal-sm', null],
            [2, 'modal-lg', null], [2, 'modal-xl', null]], [[null,
                '@fadeDown.done']], function (_v, en, $event) {
            var ad = true;
            var _co = _v.component;
            if (('@fadeDown.done' === en)) {
                var pd_0 = (_co.fadeDone($event) !== false);
                ad = (pd_0 && ad);
            }
            return ad;
        }, null, null)), (_l()(), i0.ɵted(null, ['\n\n              '])),
        (_l()(), i0.ɵeld(0, null, null, 29, 'div', [['class', 'modal-outer-wrapper']], null, null, null, null, null)),
        (_l()(), i0.ɵted(null, ['\n                  '])), (_l()(), i0.ɵeld(0, null, null, 19, 'div', [['class', 'modal-content-wrapper']], null, null, null, null, null)), (_l()(), i0.ɵted(null, ['\n                      '])), (_l()(), i0.ɵted(null, ['\n                      '])),
        i0.ɵncd(null, 0), (_l()(), i0.ɵted(null, ['\n\n                      '])),
        (_l()(), i0.ɵeld(0, null, null, 13, 'div', [['class', 'modal-content']], null, null, null, null, null)),
        (_l()(), i0.ɵted(null, ['\n                          '])), (_l()(), i0.ɵeld(0, null, null, 6, 'div', [['class', 'modal-header']], null, null, null, null, null)), (_l()(), i0.ɵted(null, ['\n                              '])), (_l()(), i0.ɵand(16777216, null, null, 1, null, View_Modal_2)), i0.ɵdid(16384, null, 0, i3.NgIf, [i0.ViewContainerRef, i0.TemplateRef], { ngIf: [0, 'ngIf'] }, null),
        (_l()(), i0.ɵted(null, ['\n                              '])), i0.ɵncd(null, 1), (_l()(), i0.ɵted(null, ['\n                          '])), (_l()(),
            i0.ɵted(null, ['\n                          '])), i0.ɵncd(null, 2), (_l()(), i0.ɵted(null, ['\n                          '])), i0.ɵncd(null, 3), (_l()(), i0.ɵted(null, ['\n                      '])), (_l()(),
            i0.ɵted(null, ['\n                  '])), (_l()(), i0.ɵted(null, ['\n                  '])), (_l()(), i0.ɵeld(0, null, null, 5, 'div', [['class', 'modal-ghost-wrapper']], null, null, null, null, null)), (_l()(), i0.ɵted(null, ['\n                      '])),
        (_l()(), i0.ɵeld(0, null, null, 0, 'div', [['class', 'modal-ghost modal-ghost-1']], [[24, '@ghostPageOneState', 0]], null, null, null, null)), (_l()(), i0.ɵted(null, ['\n                      '])),
        (_l()(), i0.ɵeld(0, null, null, 0, 'div', [['class', 'modal-ghost modal-ghost-2']], [[24, '@ghostPageTwoState', 0]], null, null, null, null)), (_l()(), i0.ɵted(null, ['\n                  '])),
        (_l()(), i0.ɵted(null, ['\n              '])), (_l()(), i0.ɵted(null, ['\n          '])), (_l()(), i0.ɵted(null, ['\n\n          '])), (_l()(),
            i0.ɵeld(0, null, null, 0, 'div', [['aria-hidden', 'true'], ['class',
                    'modal-backdrop']], [[24, '@fade', 0]], [[null, 'click']], function (_v, en, $event) {
                var ad = true;
                var _co = _v.component;
                if (('click' === en)) {
                    var pd_0 = ((_co.staticBackdrop || _co.close()) !== false);
                    ad = (pd_0 && ad);
                }
                return ad;
            }, null, null)), (_l()(), i0.ɵted(null, ['\n      ']))], function (_ck, _v) {
        var _co = _v.component;
        var currVal_4 = _co.closable;
        _ck(_v, 18, 0, currVal_4);
    }, function (_ck, _v) {
        var _co = _v.component;
        var currVal_0 = _co.skipAnimation;
        var currVal_1 = (_co.size == 'sm');
        var currVal_2 = (_co.size == 'lg');
        var currVal_3 = (_co.size == 'xl');
        _ck(_v, 4, 0, currVal_0, currVal_1, currVal_2, currVal_3);
        var currVal_5 = _co.ghostPageState;
        _ck(_v, 31, 0, currVal_5);
        var currVal_6 = _co.ghostPageState;
        _ck(_v, 33, 0, currVal_6);
        var currVal_7 = undefined;
        _ck(_v, 38, 0, currVal_7);
    });
}
export function View_Modal_0(_l) {
    return i0.ɵvid(0, [(_l()(), i0.ɵted(null, ['\n\n      '])), (_l()(), i0.ɵted(null, ['\n\n      '])), (_l()(), i0.ɵand(16777216, null, null, 1, null, View_Modal_1)), i0.ɵdid(16384, null, 0, i3.NgIf, [i0.ViewContainerRef, i0.TemplateRef], { ngIf: [0, 'ngIf'] }, null), (_l()(), i0.ɵted(null, ['\n    ']))], function (_ck, _v) {
        var _co = _v.component;
        var currVal_0 = _co._open;
        _ck(_v, 3, 0, currVal_0);
    }, null);
}
export function View_Modal_Host_0(_l) {
    return i0.ɵvid(0, [(_l()(), i0.ɵeld(0, null, null, 2, 'clr-modal', [], null, [['body', 'keyup.escape']], function (_v, en, $event) {
            var ad = true;
            if (('body:keyup.escape' === en)) {
                var pd_0 = (i0.ɵnov(_v, 2).close() !== false);
                ad = (pd_0 && ad);
            }
            return ad;
        }, View_Modal_0, RenderType_Modal)), i0.ɵprd(8704, null, i5.ScrollingService, i5.ScrollingService, [i6.DOCUMENT]), i0.ɵdid(704512, null, 0, i4.Modal, [i5.ScrollingService], null, null)], null, null);
}
export var ModalNgFactory = i0.ɵccf('clr-modal', i4.Modal, View_Modal_Host_0, { _open: 'clrModalOpen', closable: 'clrModalClosable', size: 'clrModalSize',
    staticBackdrop: 'clrModalStaticBackdrop', skipAnimation: 'clrModalSkipAnimation',
    ghostPageState: 'clrModalGhostPageState', bypassScrollService: 'clrModalOverrideScrollService',
    stopClose: 'clrModalPreventClose' }, { _openChanged: 'clrModalOpenChange', altClose: 'clrModalAlternateClose' }, ['.modal-nav', '.modal-title', '.modal-body', '.modal-footer']);
